terraform/modules/service/roles.tf

```diff
@@ -0,0 +1,24 @@
+# TODO: This may not be the best way to split files, consider in CLDC-2526
+
+resource "aws_iam_role" "task_execution" {
+  name               = "${var.prefix}-task-execution"
+  assume_role_policy = data.aws_iam_policy_document.task_assume_role.json
+}
+
+data "aws_iam_policy_document" "task_assume_role" {
+  statement {
+    effect  = "Allow"
+    actions = ["sts:AssumeRole"]
+
+    principals {
+      type        = "Service"
+      identifiers = ["ecs-tasks.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role_policy_attachment" "task_execution_managed_policy" {
+  role = aws_iam_role.task_execution.name
+  # This is an aws managed policy for ecs task execution roles
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy"
```

Comment:
More for my benefit, I've generally felt it's good to avoid using AWS managed policies in case they change anything about them in future. However this one here seems pretty ideal for what we need 🤷‍♂️. What's your general preferences around these?

Reply:
I think we generally do want to use AWS managed policies if there is one designed for our use case - precisely because if something changes about the permissions required we'll automatically get the update (e.g. if AWS adds a new permission needed to start ECS tasks). And there's less risk of getting something wrong.

This only applies if it is actually our use case rather than something tangentially related that happens to need similar permissions, but in this case it is.